libraries/aws/shadow/src/aws_iot_shadow_operation.c

```diff
@@ -792,114 +782,114 @@ AwsIotShadowError_t _AwsIotShadow_ProcessOperation( IotMqttConnection_t mqttConn
                                                 &pTopicBuffer,
                                                 &operationTopicLength );
 
-    if( status != AWS_IOT_SHADOW_SUCCESS )
+    if( status == AWS_IOT_SHADOW_SUCCESS )
     {
-        IotLogError( "No memory for Shadow operation topic buffer." );
-
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_NO_MEMORY );
+        /* Get a subscription object for this Shadow operation. */
+        status = _findSubscription( pThingName,
+                                    thingNameLength,
+                                    pTopicBuffer,
+                                    operationTopicLength,
+                                    pOperation,
+                                    &freeTopicBuffer );
     }
-
-    /* Get a subscription object for this Shadow operation. */
-    status = _findSubscription( pThingName,
-                                thingNameLength,
-                                pTopicBuffer,
-                                operationTopicLength,
-                                pOperation,
-                                &freeTopicBuffer );
-
-    if( status != AWS_IOT_SHADOW_SUCCESS )
+    else
     {
-        /* No subscription was found and no subscription could be allocated. */
-        IOT_GOTO_CLEANUP();
-    }
-
-    /* Set the operation topic name. */
-    publishInfo.pTopicName = pTopicBuffer;
-    publishInfo.topicNameLength = operationTopicLength;
+        IotLogError( "No memory for Shadow operation topic buffer." );
 
-    IotLogDebug( "Shadow %s message will be published to topic %.*s",
-                 _pAwsIotShadowOperationNames[ pOperation->type ],
-                 publishInfo.topicNameLength,
-                 publishInfo.pTopicName );
+        status = AWS_IOT_SHADOW_NO_MEMORY;
+    }
 
-    /* Set the document info if this operation is not a Shadow DELETE. */
-    if( pOperation->type != SHADOW_DELETE )
+    if( status == AWS_IOT_SHADOW_SUCCESS )
```

Comment:
A similar simplification for the `if-else` code structure that I suggested for the previous function can be made.